pkg/usernames/usernames.go

```diff
@@ -0,0 +1,19 @@
+package usernames
+
+// Response represents the result of a search request done by using the MUR name or an email address.
+// The result will specify if a user exits in the system or not , or if there are multiple matches for the given search query
+type Response struct {
+	// For is the search string that was specified in the search request url. Example: /usernames/<search string>
+	For string `json:"for"`
+	// Valid can be true or false.
+	// 	true indicates that a match was found
+	// 	false indicates that no match was found
+	Valid string `json:"valid"`
```

Comment:
and alternative name could be `Found`

Reply:
Is this needed? Isn't found the same as when `Matched` contains some results and not found is the same as when `Matched` is empty?

Should this be more about validation of the search parameter instead?

Reply:
The plan is In the first phase to support search only by username. So in this case the response will be just:
GET `/usernames/john`
```
{
  for: john
  valid: true
}
```
Later we will support search also by email id, and the response might return multiple results, the valid will be set to false in that case:

GET /usernames/john@redhat.com
```
{
  for: john@redhat.com
  valid: false
  matched: [
 	{ 
       username: johny
  	},
 	{ 
       username: john-2
  	},
 	...
  ]}
```

Reply:
ah ok, makes sense for it to be `Valid` rather than `Found` in that case

Reply:
I would just return 404 if nothing found. So we can drop `Valid/Found` completely.
And do we really need to add any sub-fields there at all? Can it be just an array?

`GET /usernames/john` - 200 OK:
```
{"john"}
```
`GET /usernames/johnnn` - 404 NOT FOUND

`GET /usernames/john@domain.com` - 200 OK:
```
{"johny", "johny"}
```

Reply:
Makes sense to me to use the http status code as an indicator for found/not found. 

But in your last example, should it be 200 or 404 (since there are multiple users found) ?
```
GET /usernames/john@domain.com - 404 OK:

{"johny", "johny"}
```

Also I would keep the field name: `{"username":"johny"}`, it gives us the option to add more fields and details about the users (let's imagine we are asked to add the family name or sso id or anything else).

Reply:
> So the validation logic should be on the client side then? eg. if the api returns multiple results then the client displays an error to users?

I would say it's up to the client (UI) to decide what to do in this case. Showing an error that there are multiple users found or show all the options to the user so he/she can pick which username should be used.

Reply:
> Also I would keep the field name: {"username":"johny"}, it gives us the option to add more fields and details about the users (let's imagine we are asked to add the family name or sso id or anything else).

We could add it as:
```
{
    {"username" : "johny"},
    {"username" : "john"}
}
```
but I don't think we will be able to add anything besides the username anytime soon. Due to PII concerns. It's unlikely that we will need that but it's possible :)
I can keep this API as simple as possible and introduce a v2 version of it in the future if we really need it. But on the other hand that variant with "username": is not a big deal to have it now either. So I don't have a strong opinion.

@MatousJobanek any thought?

Reply:
I really like this approach. Let's do that. If there is no user with the name then return 404. 
```
GET /usernames/doesnotexist - 404 NotFound:
GET /usernames/john@domain.com - 404 NotFound:
```
If there is then return the list of the matching entries - right now only one:
```
GET /usernames/johny - 200 OK:
[
    {"username" : "johny"}
]
```
later when we support emails as well, we could return more of them:
```
GET /usernames/johny@rh.com - 200 OK:
[
    {"username" : "johny"},
    {"username" : "johny2"}
]
```
we could easily extend/play with this format in the future without introducing any breaking changes with the current API

Reply:
cc @rohitkrai03